src/params/progression.jl

```diff
@@ -98,6 +127,7 @@ end
 function resample!(
   rng::AbstractRNG,
   progressions, ages::AbstractArray{T} where T <: Real,
+  genders::AbstractArray{T} where T <: Bool,
```

Comment:
```suggestion
  genders::AbstractVector{Bool},
```
`Bool` is already a concrete type (`Real` is not). This corresponds to any array of Bool. For abstract class like Real that would mean that each element can be be of different type, e.g. Int and Float mixed which would be slow